src/System.Management.Automation/engine/lang/parserutils.cs

```diff
@@ -1152,6 +1153,49 @@ internal static object LikeOperator(ExecutionContext context, IScriptExtent erro
             return resultList.ToArray();
         }
 
+        /// <summary>
+        /// Implementation of the PowerShell -matchall operator.
+        /// </summary>
+        /// <param name="context">The execution context to use.</param>
+        /// <param name="errorPosition">The position to use for error reporting.</param>
+        /// <param name="lval">Left operand.</param>
+        /// <param name="rval">Right operand.</param>
+        /// <param name="ignoreCase">Ignore case?</param>
+        /// <returns>The result of the operator.</returns>
+        internal static object MatchAllOperator(ExecutionContext context, IScriptExtent errorPosition, object lval, string rval, bool ignoreCase)
+        {
+            List<MatchCollection> matches = new List<MatchCollection>();
+            string[] text;
+            if (lval is string)
+            {
+                text = new string[]{(string)lval};
+            } 
+            else
+            {
+                text = ((IEnumerable)lval).Cast<object>()
+                                 .Select(x => x.ToString())
+                                 .ToArray();
+            }
```

Comment:
Please use the same pattern as MatchOperator() - we should support PowerShell magic conversions:
```c#
            IEnumerator list = LanguagePrimitives.GetEnumerator(lval);
            if (list == null)
            {
                string lvalString = lval == null ? string.Empty : PSObject.ToStringParser(context, lval);
…
```